cmd/package-operator-manager/bootstrap/apihelper_test.go

```diff
@@ -87,3 +87,201 @@ func TestIsPKOAvailable(t *testing.T) {
+					Status:             metav1.ConditionTrue,
+					ObservedGeneration: cp.Generation,
+				})
+			}).
+			Return(nil)
+
+		isPKOClusterPackageAvailable, err := isPKOClusterPackageAvailable(
+			context.Background(),
+			c,
+		)
+
+		require.NoError(t, err)
+		assert.True(t, isPKOClusterPackageAvailable)
+	})
+}
+
+func TestIsPKOdeploymentAvailable(t *testing.T) {
+	t.Parallel()
+
+	t.Run("not found", func(t *testing.T) {
+		t.Parallel()
+		c := testutil.NewClient()
+
+		c.On("Get",
+			mock.Anything,
+			mock.Anything,
+			mock.IsType(&appsv1.Deployment{}),
+			mock.Anything).
+			Return(errors.NewNotFound(schema.GroupResource{}, ""))
+
+		isPKODeploymentAvailable, err := isPKODeploymentAvailable(
+			context.Background(), c, "")
+		require.NoError(t, err)
+		assert.False(t, isPKODeploymentAvailable)
+	})
+
+	t.Run("not available", func(t *testing.T) {
+		t.Parallel()
+		c := testutil.NewClient()
+
+		c.On("Get", mock.Anything, mock.Anything,
+			mock.AnythingOfType("*v1.Deployment"),
+			mock.Anything).
+			Return(nil)
+
+		isPKODeploymentAvailable, err := isPKODeploymentAvailable(
+			context.Background(), c, "")
+		require.NoError(t, err)
+		assert.False(t, isPKODeploymentAvailable)
+	})
+
+	t.Run("not available - available replicas is 0", func(t *testing.T) {
+		t.Parallel()
+		c := testutil.NewClient()
+
+		c.On("Get", mock.Anything, mock.Anything,
+			mock.AnythingOfType("*v1.Deployment"),
+			mock.Anything).
+			Run(func(args mock.Arguments) {
+				depl := args.Get(2).(*appsv1.Deployment)
+				depl.Status.AvailableReplicas = 0
+			}).
+			Return(nil)
+
+		isPKODeploymentAvailable, err := isPKODeploymentAvailable(
+			context.Background(), c, "")
+		require.NoError(t, err)
+		assert.False(t, isPKODeploymentAvailable)
+	})
+
+	t.Run("not available - available replicas is not equal to available replicas", func(t *testing.T) {
+		t.Parallel()
+		c := testutil.NewClient()
+
+		c.On("Get", mock.Anything, mock.Anything,
+			mock.AnythingOfType("*v1.Deployment"),
+			mock.Anything).
+			Run(func(args mock.Arguments) {
+				depl := args.Get(2).(*appsv1.Deployment)
+				depl.Status.AvailableReplicas = 2
+				depl.Status.UpdatedReplicas = 1
+			}).
+			Return(nil)
+
+		isPKODeploymentAvailable, err := isPKODeploymentAvailable(
+			context.Background(), c, "")
+		require.NoError(t, err)
+		assert.False(t, isPKODeploymentAvailable)
+	})
+
+	t.Run("available", func(t *testing.T) {
+		t.Parallel()
+		c := testutil.NewClient()
+
+		c.On("Get", mock.Anything, mock.Anything,
+			mock.AnythingOfType("*v1.Deployment"),
+			mock.Anything).
+			Run(func(args mock.Arguments) {
+				depl := args.Get(2).(*appsv1.Deployment)
+				depl.Status.AvailableReplicas = 2
+				depl.Status.UpdatedReplicas = depl.Status.AvailableReplicas
+			}).
+			Return(nil)
+
+		isPKODeploymentAvailable, err := isPKODeploymentAvailable(
+			context.Background(), c, "")
+		require.NoError(t, err)
+		assert.True(t, isPKODeploymentAvailable)
+	})
+}
```

Comment:
I might be wrong but this to me looks identical to https://github.com/package-operator/package-operator/blob/7186448b9e86a9920496b3d3155c8e8e117e5669/cmd/package-operator-manager/bootstrap/apihelper_test.go#L20-L89

Reply:
Here as the name suggests, we are testing for `isPKODeploymentAvailable` which is missing. The snippet you are referring is for `isPKOAvailable`. 🙂

Reply:
`isPKOAvailable` is internally calling both `isPKOClusterPackageAvailable` and `isPKODeploymentAvailable`.

I think these tests could be rewritten by just calling `isPKOAvailable` including all the possible test cases **only once** (the first two test cases in `TestIsPKOAvailable` and `TestIsPKODeploymentAvailable` are _exactly_ the same).

The coverage would be the same but the readability would drastically improve.

Reply:
Done.

Reply:
ok looks better now